src/main/java/dev/codesoapbox/dummy4j/DefaultExpressionResolver.java

```diff
@@ -88,25 +81,63 @@ private String replace(String expression, Matcher expressionMatcher, Supplier<St
     }
 
     /**
-     * Resolves a single key to a random value.
+     * Resolves a single path to a random value from a superset of all locale values.
+     * Note that this method does not perform any parsing and thus will not resolve values which themselves are
+     * expressions.
+     *
+     * @param path the path to resolved
+     * @return a random value based on the path
+     */
+    private String resolvePathWithinAllLocales(String path) {
+        List<String> result = locales.stream()
+                .map(locale -> localizedDefinitions.get(locale).resolve(path))
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .distinct()
+                .collect(toList());
+
+        if (!result.isEmpty()) {
+            return getRandom(result);
+        }
+        return "";
+    }
+
+    private String getRandom(List<String> result) {
+        int i = randomService.nextInt(result.size() - 1);
+        return result.get(i);
+    }
+
+    private String replaceSingleLocalePlaceholders(String expression) {
+        final Matcher expressionMatcher = SINGLE_LOCALE_VARIABLE_PATTERN.matcher(expression);
+
+        return replace(expression, expressionMatcher,
+                () -> resolvePathWithinSingleLocale(expressionMatcher.group(1)));
+    }
+
+    /**
+     * Resolves a single path to a random value from a single locale.
+     * The method first looks into the primary locale and then goes to the next ones in order if a path could not be
```

Comment:
`goes to the next ones in order if a path could` - not sure if understand grammar here: in order if sth?
Maybe: The method checks the primary locale first and then moves on to the next ones if the path cannot be resolved.

Reply:
It "goes to the next one", preserving order. Perhaps "preserving order" would be more readable here?